pkg/operator/nodekubeconfigcontroller/nodekubeconfigcontroller.go

```
@@ -0,0 +1,181 @@
+	"github.com/openshift/library-go/pkg/operator/resource/resourceread"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/kubernetes"
+	coreclientv1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	corev1listers "k8s.io/client-go/listers/core/v1"
+)
+
+const workQueueKey = "key"
+
+type NodeKubeconfigController struct {
+	operatorClient v1helpers.StaticPodOperatorClient
+
+	kubeClient          kubernetes.Interface
+	configMapLister     corev1listers.ConfigMapLister
+	secretLister        corev1listers.SecretLister
+	infrastuctureLister configv1listers.InfrastructureLister
+}
+
+func NewNodeKubeconfigController(
+	operatorClient v1helpers.StaticPodOperatorClient,
+	kubeInformersForNamespaces v1helpers.KubeInformersForNamespaces,
+	kubeClient kubernetes.Interface,
+	infrastuctureInformer configv1informers.InfrastructureInformer,
+	eventRecorder events.Recorder,
+) factory.Controller {
+	c := &NodeKubeconfigController{
+		operatorClient:      operatorClient,
+		kubeClient:          kubeClient,
+		configMapLister:     kubeInformersForNamespaces.ConfigMapLister(),
+		secretLister:        kubeInformersForNamespaces.SecretLister(),
+		infrastuctureLister: infrastuctureInformer.Lister(),
+	}
+
+	return factory.New().WithInformers(
+		operatorClient.Informer(),
+		kubeInformersForNamespaces.InformersFor(operatorclient.OperatorNamespace).Core().V1().ConfigMaps().Informer(),
+		kubeInformersForNamespaces.InformersFor(operatorclient.TargetNamespace).Core().V1().ConfigMaps().Informer(),
+		kubeInformersForNamespaces.InformersFor(operatorclient.OperatorNamespace).Core().V1().Secrets().Informer(),
+		kubeInformersForNamespaces.InformersFor(operatorclient.TargetNamespace).Core().V1().Secrets().Informer(),
+		infrastuctureInformer.Informer(),
+	).WithSync(c.sync).ResyncEvery(time.Second).ToController("NodeKubeconfigController", eventRecorder.WithComponentSuffix("node-kubeconfig-controller"))
```

use the withdegraded option so you can just return an error and get degraded automatically.

---

done